pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -62,6 +65,31 @@ func newConntrackRoundTripper(name string, transport *http.Transport) *http.Tran
 	return transport
 }
 
+func httpLoggerMiddleware() httpclient.Middleware {
```

Comment:
Sorry a bit late to the party. Where's the parsing of the plugin settings happening currently so they end up in `opts.Lables["har_log_enabled"]`? To me this middleware is for core plugins so would make more sense to use the settings from .ini directly in here or via setting package. 

In addition a nit, please move this middleware to a separate file.

Reply:
@marefr i was actually trying to figure that out with this new approach. i'll move this to a new file and give that a shot. thanks for the suggestion!